File: backend/electron-ecommerce-rest/src/main/java/com/electron/rest/dto/order/OrderResponse.java
package com.electron.rest.dto.order;

import com.electron.rest.entity.orders.OrderItem;
import lombok.Builder;

import java.util.List;

@Builder
public record OrderResponse(
        String status,
        Integer totalItems,
        Double totalPrice,
        String fullName,
        String streetOne,
        String streetTwo,
        String city,
        String state,
        String postcode,
        List<OrderItemDto> orderItems
) {
}

File: frontend/electron-ecommerce-react/src/api/dto/order/OrderResponse.ts
import {OrderItemDto} from "./OrderItemDto.ts";

export type OrderResponse = {
    status: string,
    totalItems: number,
    totalPrice: number,
    fullName: string,
    streetOne: string,
    streetTwo: string,
    city: string,
    state: string,
    postcode: string,
    orderItems: OrderItemDto[],
}